js/Common.js

```diff
@@ -26,6 +26,19 @@ function SetupArrayRandomFunction(){
 			}
 }
 
+// Return a random element of the array if it's not equal to previuousValue
+function SetupArrayNonRepeatingRandomFunction(){
+	Array.prototype.notRepeatingRandom = function (previousValue) {
+		var result = this[Math.floor((Math.random()*this.length))];
+		if (result != previousValue) return result;
+			else {
+				result = this[Math.floor((Math.random()*this.length))];
+				if (result != previousValue) return result;
```

Comment:
Even the second random value may match the previousOne. 
Probably while loop would be better. Let's say if 10 times we get the same values, return the next value of the array.